### DISCUSSION TITLE
bin/hex mode

### DISCUSSION THREAD

**Discussion by msasdk** [sourcelair/xterm.js#4420] [Q&A] [answered]
Hello!

Can xterm.js be set to non ascii mode (bin/hex...) similar to this?

![image](https://user-images.githubusercontent.com/13538204/220458147-5c9ff806-0bb8-4bc4-83b4-c126c95435b3.png)

**Comment by jerch** [comment, marked as answer]
Nope, thats not possible out of the box. xterm.js is not even in ASCII mode, but in UTF8. You still can achieve that behavior by routing incoming data through a byte --> hex converter (either on system side, or in JS).

**Reply by msasdk** [reply]
Thanks!
